common/djangoapps/terrain/stubs/tests/test_edxnotes.py

```diff
@@ -139,24 +149,110 @@ def test_update(self):
         response = requests.get(self._get_url("api/v1/annotations/does_not_exist"))
         self.assertEqual(response.status_code, 404)
 
+    # pylint: disable=too-many-arguments
+    def _verify_pagination_info(
+            self,
+            response,
+            total_notes,
+            num_pages,
+            notes_per_page,
+            current_page,
+            previous_page,
+            next_page,
+            start
+    ):
+        """
+        Verify the pagination information.
+
+        Argument:
+            response: response from api
+            total_notes: total notes in the response
+            num_pages: total number of pages in response
+            notes_per_page: number of notes in the response
+            current_page: current page number
+            previous_page: previous page number
+            next_page: next page number
+            start: start of the current page
+        """
+        def get_page_value(url, current_page):
+            """
+            Return page value extracted from url.
+            """
+            if url == "null":
+                return None
+
+            parsed = urlparse.urlparse(url)
+            query_params = urlparse.parse_qs(parsed.query)
+
+            # If current_page is 2 then DRF will not include `page` query param in previous url.
+            # So return page 1 if current page equals to 2 and `page` key is missing from url.
+            if "page" not in query_params and current_page == 2:
```

Comment:
This will never execute because in stub server page param is always added even if current_page is 2.